### ISSUE TITLE
How to train this network on new dataset?

### ISSUE THREAD

**Issue by guopei** [anewell/pose-hg-train#15]
Hi @anewell 

Thank you so much for sharing this training code. I am wondering if there is a tutorial for training on a new dataset. For example the cub-200-2011 birds dataset. It has provided key points locations and bounding box annotations. Are these enough for making the network run? Thanks!

**Comment by anewell** [comment]
Hi,

To have this run on a new dataset you can create a file similar to src/util/dataset/mpii.lua and then run with the additional command "-dataset your-new-dataset". Hopefully it shouldn't be too much trouble to adapt the necessary functions.

For example, most of the relevant information to generate a sample is passed through "getPartInfo" which returns the keypoint locations of the target person, along with a coordinate for the center of the person and a scale value. You can adapt the bounding box info from your dataset to provide an appropriate center and scale.

Let me know if you have any questions about getting it set up!

- Alejandro

**Comment by guopei** [comment]
Hi, Alejandro.

Thank you for your timely reply. I now get the model working on my own dataset. But another thing I was confused about is I don't see the training converge. I tried several learning rate, 2.5e-4, 1e-3, 1e-2, etc, all of them will give nearly 0.0% accuracy when testing after several epochs, and the key point output remains constant no matter what the input is. I know there must be something wrong, but I have no clue where. Can you shed some light onto it? I am using the pre-trained model on human and fine tune it on bird dataset.  Thanks!

- Pei

**Comment by guopei** [comment]
Sorry, that's caused by my own implementation bug... During my debugging, I set all input to be zero and forgot to set them back...
